tests/system/program-execution-test.cpp

```diff
@@ -0,0 +1,253 @@
+      ASSERT_EQ(0, value) << "In Register " << reg;
+    });
+    this->executeAll();
+    this->waitUntilExecutionFinished();
+    this->assertForAllRegisters([](auto reg, SizeType value) {
+      if (reg != "x1") {
+        EXPECT_EQ(0, value) << "In Register " << reg;
+      }
+    });
+    this->assertRegisterValue("x1", 479001600);
+  }
+};
+
+using FactorialTest32 = FactorialTest<int32_t>;
+
+TEST_F(FactorialTest32, factorial32) { factorialTest(); }
+
+using FactorialTest64 = FactorialTest<int64_t>;
+
+TEST_F(FactorialTest64, factorial64) { factorialTest(); }
+
+template <typename SizeType>
+struct FactorialRecTest : public ProgramExecutionFixture<SizeType, true, 1024> {
+  FactorialRecTest()
+      : ProgramExecutionFixture<SizeType, true, 1024>("factorial_rec.txt") {}
+  void factorialRecTest() {
+    this->assertForAllRegisters([](auto reg, auto value) {
+      ASSERT_EQ(0, value) << "In Register " << reg;
+    });
+    this->executeAll();
+    this->waitUntilExecutionFinished();
+    this->assertRegisterValue("x5", 479001600);
+  }
+};
+
+using FactorialRecTest32 = FactorialRecTest<int32_t>;
+TEST_F(FactorialRecTest32, factorialRec32) { factorialRecTest(); }
+
+using FactorialRecTest64 = FactorialRecTest<int64_t>;
+TEST_F(FactorialRecTest64, factorialRec64) { factorialRecTest(); }
+
+template <typename SizeType>
+struct MemoryIOTest : public ProgramExecutionFixture<SizeType, false, 1024> {
+  MemoryIOTest()
+      : ProgramExecutionFixture<SizeType, false, 1024>("memoryio.txt") {}
+  void memoryioTest() {
+    this->assertForAllRegisters([](auto reg, auto value) {
+      ASSERT_EQ(0, value) << "In Register " << reg;
+    });
+    this->executeAll();
+    this->waitUntilExecutionFinished();
+    this->assertRegisterValue("x1", 0x489);
+    this->assertMemory(0, 4, 0x489);
+  }
+};
+
+using MemoryIOTest32 = MemoryIOTest<int32_t>;
+TEST_F(MemoryIOTest32, memoryio32) { memoryioTest(); }
+
+using MemoryIOTest64 = MemoryIOTest<int64_t>;
+TEST_F(MemoryIOTest64, memoryio64) { memoryioTest(); }
+
+template <typename SizeType>
+struct SuperSumTest : public ProgramExecutionFixture<SizeType> {
+  SuperSumTest() : ProgramExecutionFixture<SizeType>("super_sum.txt") {}
+
+  void superSumTest() {
+    this->assertForAllRegisters(
+        [](auto reg, auto val) { ASSERT_EQ(0, val) << "In Register " << reg; });
+    this->executeAll();
+    this->waitUntilExecutionFinished();
+    this->assertRegisterValue("x5", 35);//1+(1+2)+(1+2+3)+(1+2+3+4)+(1+2+3+4+5)
+    // = 1+3+6+10+15 = 35
+  }
+};
+
+using SuperSumTest32 = SuperSumTest<int32_t>;
+TEST_F(SuperSumTest32, supersum32) { superSumTest(); }
+
+using SuperSumTest64 = SuperSumTest<int64_t>;
+TEST_F(SuperSumTest64, supersum64) { superSumTest(); }
+
+template <typename SizeType>
+struct EndlessTest : public ProgramExecutionFixture<SizeType> {
+  EndlessTest() : ProgramExecutionFixture<SizeType>("endless.txt") {}
+
+  void endlessTest() {
+    this->assertForAllRegisters(
```

Comment:
No need for `this->` here. If anything add `super = ProgramExecutionFixture<SizeType>` and write `super::assertForAllRegisters` to indicate that this method is declared by the base class